sklearn/metrics/regression.py

```diff
@@ -181,6 +183,46 @@ def mean_absolute_error(y_true, y_pred,
     return np.average(output_errors, weights=multioutput)
 
 
+def mean_absolute_percentage_error(y_true, y_pred):
+    """Mean absolute percentage error regression loss
+
+    Read more in the :ref:`User Guide <mean_absolute_percentage_error>`.
+
+    Parameters
+    ----------
+    y_true : array-like of shape = (n_samples)
+        Ground truth (correct) target values.
+
+    y_pred : array-like of shape = (n_samples)
+        Estimated target values.
+
+    Returns
+    -------
+    loss : float
+        A positive floating point value (the best value is 0.0).
+
+    Examples
+    --------
+    >>> from sklearn.metrics import mean_absolute_percentage_error
+    >>> y_true = [3, -0.5, 2, 7]
+    >>> y_pred = [2.5, 0.0, 2, 8]
+    >>> mean_absolute_percentage_error(y_true, y_pred)
+    32.738...
+    """
+    y_type, y_true, y_pred, _ = _check_reg_targets(y_true, y_pred,
+                                                   'uniform_average')
+
+    if y_type == 'continuous-multioutput':
+        raise ValueError("Multioutput not supported "
+                         "in mean_absolute_percentage_error")
+
+    if (y_true == 0).any():
+        raise ValueError("mean_absolute_percentage_error requires"
```

Comment:
This is not currently executed in any tests. It should be tested

Reply:
I added a test case called `test_raise_value_error_y_with_zeros` for this particular error